### PR TITLE
Jptom hw2 attempt1

### PR DESCRIPTION
**Pull request by jptom**
---
A good PR should include the following parts:

- [x] A clear title
- [x] A meaningful message for PR, as well as its commits
- [x] from your specific branch (***not master***) merging to teacher's `HW` branch (***not master***, again)
- [x] excluding any irrelevant files, such as binaries, text files, or dot files
- [x] passing tests/CI

### CODE REVIEW

File: hw2/jptom/q2/test_q2.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python
+import _vector
+import numpy as np
+import numpy.testing as npt
+
+def transform_2d(vec, theta):
```
Comment (tigercosmos) [inline hw2/jptom/q2/test_q2.py]: cool

File: hw2/jptom/q2/vector.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+	if ( vec1_info.ndim != 1 || vec2_info.ndim != 1 ) { throw std::invalid_argument("invalid input. input vector ndidm must be 1."); }
+
+	if ( vec1_info.size != 2 || vec2_info.size != 2 ) { throw std::invalid_argument("invalid input. input vector size must be 2."); }
+
+	double * data1 = (double *) vec1_info.ptr;
+	double * data2 = (double *) vec2_info.ptr;
+	if ( (data1[0]==0 && data1[1]==0) || (data2[0]==0 && data2[1]==0) ) { throw std::invalid_argument("invalid input. input vector has zero length."); }
+	//
+	//std::cout<<"vec1:[ "<<data1[0]<<", "<<data1[1]<<"]"<<std::endl;
```
Comment (tigercosmos) [inline hw2/jptom/q2/vector.cpp]: It's a good habit to remove useless code when you push your code

File: hw2/jptom/q1/line.hpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	public:
+		Line();
+		Line(Line const & other) : m_size(other.m_size), m_coord(other.m_coord) {}
+		Line(Line      && other) : m_size(other.m_size), m_coord(move(other.m_coord)) {}
+		Line & operator=(Line const & );
+		Line & operator=(Line       &&);
+		Line(size_t size) { m_size = size; for (size_t t=0; t<m_size*2; t++) { m_coord.push_back(0); }}
+		~Line() {}
+		size_t size() const { return m_size; }
+		int   x(size_t it) const { check_range(it); return m_coord[it*2  ]; }
```
Comment (tigercosmos) [inline hw2/jptom/q1/line.hpp]: this should be `float` or `double`

Comment (jptom) [reply hw2/jptom/q1/line.hpp]: I modified.

**Pull request merged**